src/engine/QueryPlanner.cpp

```diff
@@ -1159,17 +1161,21 @@ Variable QueryPlanner::generateUniqueVarName() {
 }
 
 // _____________________________________________________________________________
-QueryPlanner::SubtreePlan QueryPlanner::getTextLeafPlan(
+vector<QueryPlanner::SubtreePlan> QueryPlanner::getTextLeafPlan(
     const QueryPlanner::TripleGraph::Node& node) const {
-  SubtreePlan plan(_qec);
-  plan._idsOfIncludedNodes |= (size_t(1) << node._id);
-  auto& tree = *plan._qet;
+  vector<SubtreePlan> vecPlans;
   AD_CONTRACT_CHECK(node._wordPart.has_value());
-  auto textOp = std::make_shared<TextOperationWithoutFilter>(
-      _qec, node._wordPart.value(), node._variables, node._cvar.value());
-  tree.setOperation(QueryExecutionTree::OperationType::TEXT_WITHOUT_FILTER,
-                    textOp);
-  return plan;
+  for (string word : node._wordPart.value()) {
+    SubtreePlan plan(_qec);
+    plan._idsOfIncludedNodes |= (size_t(1) << node._id);
+    auto& tree = *plan._qet;
+    auto textOp = std::make_shared<WordIndexScan>(_qec, node._variables,
+                                                  node._cvar.value(), word);
+    tree.setOperation(QueryExecutionTree::OperationType::WORD_INDEX_SCAN,
+                      textOp);
```

Comment:
I think we have a `makeSubtreePlan` method that is used inside the QueryPlanner to not always have to write this boilerplate here.